askama_derive/src/generator/filter.rs

```diff
@@ -516,6 +515,60 @@ impl<'a> Generator<'a, '_> {
         buf.write(")?");
         Ok(DisplayWrap::Unwrapped)
     }
+
+    fn visit_center_filter(
+        &mut self,
+        ctx: &Context<'_>,
+        buf: &mut Buffer,
+        args: &[WithSpan<'a, Expr<'a>>],
+        node: Span<'_>,
+    ) -> Result<DisplayWrap, CompileError> {
+        self.visit_filter_with_length(ctx, buf, args, node, "center")
+    }
+
+    fn visit_truncate_filter(
+        &mut self,
+        ctx: &Context<'_>,
+        buf: &mut Buffer,
+        args: &[WithSpan<'a, Expr<'a>>],
+        node: Span<'_>,
+    ) -> Result<DisplayWrap, CompileError> {
+        self.visit_filter_with_length(ctx, buf, args, node, "truncate")
+    }
+
+    fn visit_filter_with_length(
+        &mut self,
+        ctx: &Context<'_>,
+        buf: &mut Buffer,
+        args: &[WithSpan<'a, Expr<'a>>],
+        node: Span<'_>,
+        name: &str,
+    ) -> Result<DisplayWrap, CompileError> {
+        ensure_filter_has_feature_alloc(ctx, name, node)?;
```

Comment:
I don't think checking this here is a good idea. What happens if we add more filters that require `alloc` but don't have a `length` argument? Imo this is becoming too specific and not really a good idea for future developments.

Reply:
All filters in `BUILTIN_FILTERS` actually need `alloc`. :D I added a commit to fix that: <https://github.com/askama-rs/askama/pull/403/commits/d14126f53628d55edcd30a1c84d3d860368b341f>

Reply:
Oh convenient :D